org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -102,6 +104,7 @@ protected IStatus run(IProgressMonitor monitor) {
 
 	public ProjectsManager(PreferenceManager preferenceManager) {
 		this.preferenceManager = preferenceManager;
+		this.mavenBuildSupport = new MavenBuildSupport(MavenPlugin.getProjectConfigurationManager());
```

Comment:
-1. I'd like to decouple ProjectsManager from underlying build supports eventually. I expect buildSupports() implementation to load IBuildSupport instances from extension points, and don't want to make any assumption about the statefulness of each IBuildSupport.

Reply:
If you're creating a new instance every time you need one, you ARE making an assumption, namely that build support is stateless. Keeping the same instance allows the implementers to be stateless or not.